peas/tests/serial_handlers/protocol_arduino_simulator.py

```diff
@@ -0,0 +1,422 @@
+"""Provides a simple simulator for telemetry_board.ino or camera_board.ino."""
+
+import datetime
+import json
+import queue
+import random
+from serial import serialutil
+import sys
+import threading
+import time
+
+from pocs.tests import serial_handlers
+import pocs.utils.logger
+
+
+def _drain_queue(q):
+    cmd = None
+    while not q.empty():
+        cmd = q.get_nowait()
+    return cmd  # Present just for debugging.
+
+
+class ArduinoSimulator:
+    """Simulates the serial behavior of the PANOPTES Arduino sketches.
+
+    The RS-232 connection is simulated with an input and output queue of bytes. This class provides
+    a run function which can be called from a Thread to execute. Every two seconds while running it
+    will generate another json output line,
+    at around 1000 bytes per second (near 9600 buad). The bytes  (actually chunked up in batches, so not emitting a single byte
```

Comment:
`buad` -> `baud`. And the last sentence just needs some correction.

Reply:
Done.